smtcomp/defs.py

```diff
@@ -1212,8 +1212,8 @@ def get(self, d: None | dict[Track, dict[Division, set[Logic]]] = None) -> dict[
         return d
 
     def complete(self, archive: Archive | None, command: Command | None) -> ParticipationCompleted:
-        archive = cast(Archive, archive if self.archive is None else self.archive)
-        command = cast(Command, command if self.command is None else self.command)
+        archive = cast(Archive, archive if self.archive is None and self.aws_dockerfile is None else self.archive)
+        command = cast(Command, command if self.command is None and self.aws_dockerfile is None else self.command)
         return ParticipationCompleted(
             tracks=self.get(), archive=archive, command=command, experimental=self.experimental
         )
```

Comment:
Are you sure this wouldn't cause more problems? Wouldn't it be better to find a way to change the completeness criterion so one can complete participations for those two tracks two?

Reply:
I have changed it so completion now throws an error for the aws tracks. Moreover, I have changed aws_dockerfile to aws_repository because I checked the rules and apparently they demand a link to a GitHub repository (and some solvers seem to have different dockerfiles for master and servant). And I added a template submission for the cloud and parallel tracks.